### DISCUSSION TITLE
The IPsec tunnel fails to re-establish when the connection with the IPsec server is lost.

### DISCUSSION THREAD

**Discussion by muhammadi123** [strongswan/strongswan#2144] [General]
Hi,
I'm facing this issue IPsec tunnel is not getting re-establish when IPsec server connection is down and up again.

we are using below strongswan version:
strongSwan 5.3.2, Linux 3.10.84-perf, armv7l

Here the IPsec.conf configuration 
# cat /opt/strongswan/etc/ipsec.conf 
# This file is generated by Enb
config setup
        strictcrlpolicy=no
        charondebug="ike 1, knl 1, cfg 1, net 1, esp 1, dmn 1, mgr 1"

conn %default
        keyexchange=ikev2
        fragmentation=yes
        auto=add
        keyingtries=5
        ikelifetime=864000
        keylife=86400
        rekeymargin=540
        rekeyfuzz=0%
        ike=aes256-sha1-modp2048, aes256-sha256-modp2048, aes192-sha1-modp2048, aes192-sha256-modp2048, aes128-sha1-modp2048, aes128-sha256-modp2048, 3des-sha1-modp2048, 3des-sha256-modp2048!
        esp=aes256-sha1-modp2048, aes256-sha256-modp2048, aes192-sha1-modp2048, aes192-sha256-modp2048, aes128-sha1-modp2048, aes128-sha256-modp2048, 3des-sha1-modp2048, 3des-sha256-modp2048!
        dpdaction=clear
        dpddelay=30
        leftsourceip=%config6
        reauth=no

conn airspan
        mobike=no
        left=a.a.a.a
        leftid=xxxx.airspan
        leftcert=zzz.cer
        right=b.b.b.b
        rightid=*
        rightca="C=US, O=Airspan Networks Inc., CN=Airspan Networks Inc. 2048-bit Root CA"
        rightikeport=500
        rightsubnet=::/0
        type=tunnel
        auto=add
        leftupdown=/bs/bin/updownScript.sh

conn S1_1
        keyexchange=ikev2
        auto=add
        keyingtries=5
        ikelifetime=864000
        keylife=86400
        rekeymargin=540
        rekeyfuzz=0%
        dpdaction=clear
        dpddelay=30
        leftsourceip=%config6
        reauth=no

        mobike=no
        ike=aes256-sha1-modp2048, aes256-sha256-modp2048, aes192-sha1-modp2048, aes192-sha256-modp2048, aes128-sha1-modp2048, aes128-sha256-modp2048, 3des-sha1-modp2048, 3des-sha256-modp2048!
        esp=aes256-sha1, aes256-sha256, aes192-sha1, aes192-sha256, aes128-sha1, aes128-sha256, 3des-sha1, 3des-sha256!
        left=a.a.a.a
        leftid=xxxx.airspan
        leftcert=zzz.cer
        right=b.b.b.b
        rightid=*
        rightca="C=US, O=Airspan Networks Inc., CN=Airspan Networks Inc. 2048-bit Root CA"
        rightikeport=500
        rightsubnet=::/0
        type=tunnel
        auto=add
        leftupdown=/bs/bin/multiIpsecUpDownScript.sh
conn S1_2
        keyexchange=ikev2
        auto=add
        keyingtries=5
        ikelifetime=864000
        keylife=86400
        rekeymargin=540
        rekeyfuzz=0%
        dpdaction=clear
        dpddelay=30
        leftsourceip=%config6
        reauth=no

        mobike=no
        ike=aes256-sha1-modp2048, aes256-sha256-modp2048, aes192-sha1-modp2048, aes192-sha256-modp2048, aes128-sha1-modp2048, aes128-sha256-modp2048, 3des-sha1-modp2048, 3des-sha256-modp2048!
        esp=aes256-sha1, aes256-sha256, aes192-sha1, aes192-sha256, aes128-sha1, aes128-sha256, 3des-sha1, 3des-sha256!
        left=a.a.a.a
        leftid=xxxx.airspan
        leftcert=zzz.cer
        right=b.b.b.b
        rightid=*
        rightca="C=US, O=Airspan Networks Inc., CN=Airspan Networks Inc. 2048-bit Root CA"
        rightikeport=500
        rightsubnet=::/0
        type=tunnel
        leftupdown=/bs/bin/multiIpsecUpDownScript.sh
        ?"
        
        Here is the strongswan.conf 
        # strongswan.conf - strongSwan configuration file
#
# Refer to the strongswan.conf(5) manpage for details
#
# Configuration changes should be made in the included files

charon {
	load_modular = yes
	charon.prefer_configured_proposals = no
	plugins {
		openssl {
			fips_mode = 0
		}
		include strongswan.d/charon/*.conf
	}
}

include strongswan.d/*.conf

Please help us resolve this issue.
Thanks ,
Muhammadi

**Comment by tobiasbrunner** [comment]
> strongSwan 5.3.2

That's a very old version. I wouldn't recommend using it.

> when IPsec server connection is down and up again

What exactly does that mean? How does it go down and up again? Is it a local change (interface, routing, addresses), an intermediate network issue, or is the server explicitly closing the connection?

Anyway, you might want to read the docs and check options such as `auto=route`, `dpd_action` and `close_action`.

**Reply by muhammadi123** [reply]
Hi ,
What exactly does that mean? How does it go down and up again? Is it a local change (interface, routing, addresses), an intermediate network issue, or is the server explicitly closing the connection?
Multiple causes which is making connection down Interface network issue, strongswan server down .

I have added below option with this I see when connection is down and up IPsec tunnel sometimes comes up but some times still  failing with this error "no private key found for 'cccccccc.airspan.com'"
 cat /opt/strongswan/etc/ipsec.conf 	
	        auto=route
		dpdaction=restart
		closeaction=restart


initiating IKE_SA S1_1[10] to fc74:x:xx:x::253
2024-03-06T12:18:13.128003+00:00 fc74:x:x:x:x::35 charon: 09[CFG] received stroke: initiate 'S1_1'
2024-03-06T12:18:13.128736+00:00 fc74:x:x:x:x::35 charon: 14[IKE] initiating IKE_SA S1_1[10] to fc74:x:x:x::253
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) ]
sending packet: from fc74:x:x:x::14b[500] to fc74:x:x:x::253[500] (756 bytes)
received packet: from fc74:x:x:x::253[500] to fc74:x:x:x::14b[500] (481 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(HASH_ALG) N(MULT_AUTH) ]
received cert request for "C=US, O=Airspan Networks Inc., CN=Airspan Networks Inc. 2048-bit Root CA"
sending cert request for "C=US, O=Airspan Networks Inc., CN=Airspan Networks Inc. 2048-bit Root CA"
no private key found for 'cccccccc.airspan.com'
establishing connection 'S1_1' failed
initiating IKE_SA S1_2[11] to fc74:x:x:x::10
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) ]
sending packet: from fc74:x:x:x::14b[500] to fc74:x:x:x::10[500] (756 bytes)
received packet: from fc74:x:x:x::10[500] to fc74:x:x:x::14b[500] (529 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(HASH_ALG) N(MULT_AUTH) ]
received cert request for "C=US, O=Airspan Networks Inc., CN=Airspan Networks Inc. 2048-bit Root CA"
received 2 cert requests for an unknown ca
sending cert request for "C=US, O=Airspan Networks Inc., CN=Airspan Networks Inc. 2048-bit Root CA"
no private key found for 'cccccccc.airspan.com'
establishing connection 'S1_2' failed

Thanks,
Muhammadi

**Reply by tobiasbrunner** [reply]
No idea, maybe you configured something incorrectly (e.g. certificate not matching the private key, or incorrect identities).

Ah, and depending on the traffic patterns, `auto=route` causes the SA to get reestablished even without setting the other options to `restart` (could also be forced by generating traffic regularly). In your old version this combination might even cause lots of duplicate SAs.

**Reply by muhammadi123** [reply]
When I give this command manually then tunnel's are coming up
/opt/strongswan/sbin/ipsec restart

**Reply by tobiasbrunner** [reply]
Again, no idea. You are using a 6 year old version, so you're on your own with that.
